cli/pkg/cli/cli.go

```diff
@@ -39,6 +40,7 @@ const (
 type CLI struct {
 	state.Manager
 	presenter.Presenter
+	updater.Updater
```

Comment:
I don't think there is a need for a new concept here, I think almost everything can be hidden inside the existing state.Manager and presenter.Presenter concepts, if we create a new "CICDStateManagerPresenter" which wraps the VMClarity state manager and presenter, the "WaitForVolumeAttachment" and "IsAborted" functions can be empty.

"MarkInProgress" can:
- Create the Asset as Defined by the user
- Create a Scan with just that Asset defined as the scanned asset
- Create a ScanResult with that Scan ID
- Call the VMClarity state manager MarkInProgress with the ScanResult ID

then "MarkDone" can:
- Call the normal VMClarity state manager MarkDone
- Update the Scan with the summary from the ScanResult and mark it as completed.

All the presenter functions will be passed straight through except the ScanResult ID from MarkInProgress will be injected. This way the work that Erez is doing to allow for incremental status updates etc will still work even in the CI/CD flow.

If we don't need the incremental updates stuff and we don't want a Scan in the system until we've succeeded, we can leave MarkInProgress empty, and then cache all the results as they arrive through our presenter interface, and then do everything we need to do in the "MarkDone".

Reply:
The update was removed from the cli